src/inc/msquic.hpp

```diff
@@ -765,14 +782,42 @@ struct MsQuicListener {
 
     QUIC_STATUS GetInitStatus() const noexcept { return InitStatus; }
     bool IsValid() const { return QUIC_SUCCEEDED(InitStatus); }
-    MsQuicListener(MsQuicListener& other) = delete;
-    MsQuicListener operator=(MsQuicListener& Other) = delete;
+    MsQuicListener(const MsQuicListener& Other) = delete;
+    MsQuicListener& operator=(const MsQuicListener& Other) = delete;
     operator HQUIC () const noexcept { return Handle; }
-};
 
-enum MsQuicCleanUpMode {
-    CleanUpManual,
-    CleanUpAutoDelete,
+    static
+        QUIC_STATUS
+        QUIC_API
+        NoOpCallback(
+            _In_ MsQuicListener* /* Listener */,
+            _In_opt_ void* /* Context */,
+            _Inout_ QUIC_LISTENER_EVENT* /* Event */
+            ) noexcept {
+            return QUIC_STATUS_SUCCESS;
+        }
```

Comment:
Fix formatting and we need to return an error by default, otherwise it would just accept a connection and leak it.
```suggestion
    static
    QUIC_STATUS
    QUIC_API
    NoOpCallback(
        _In_ MsQuicListener* /* Listener */,
        _In_opt_ void* /* Context */,
        _Inout_ QUIC_LISTENER_EVENT* /* Event */
        ) noexcept {
        return QUIC_STATUS_NOT_SUPPORTED;
    }
```

Reply:
In fact, I'm not sure it ever makes sense to have a no-op callback for a listener. Maybe we just remove it and always require a callback.

Reply:
Makes sense. I will remove this an also update the default parameters in the MsQuicListener c'tor.